components/forms/src/api.rs

```diff
@@ -0,0 +1,248 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::*;
+use rusqlite::{Connection, NO_PARAMS, Transaction};
+use serde::Serialize;
+use serde_derive::*;
+use sql_support::{self, ConnExt};
+use std::time;
+use sync_guid::Guid;
+
+#[derive(Debug, Clone, Hash, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Address {
+    #[serde(rename = "id")]
+    pub guid: Guid,
+
+    pub given_name: String,
```

Comment:
If the database columns are nullable, these could be `Option<String>`.

Reply:
But why is that a good thing? :)

Reply:
When I wrote this at first, I was thinking you'd need to handle the field not existing—so if it was `Option<String>`, you'd need to remember to check `Some` and `None`, but with a regular `String`, you might forget to check if `given_name.is_empty()`. But like you said yesterday, you could still end up with `Some("")`, and it makes the API so clunky to handle it that it's not a good thing 😅